### PR TITLE
Fixed boxchecked value set incorrect.

### PR DESCRIPTION
**Pull request by roland-d**
## Problem

In the list forms where the hidden input value boxchecked is used, it has a default value of 0, no checkboxes are checked. Once you check a box the value of this field is changed to 01 instead of 1. When you uncheck the box, the value becomes 01-1 instead of 2 and it goes on from there.
## Solution

The value must be cast to an integer for JavaScript to be able to calculate with it.
## Testing

To test this issue, you must use the Debug console of the browser to check the initial value of the boxchecked field.
1. Go to the article list
2. Check the initial value of the boxchecked field is 0
3. Click on a checkbox to select it
4. The value becomes 01
5. Uncheck the box and the value becomes 01-1
6. Apply the patch
7. Reload the page
8. Check the initial value of the boxchecked field is 0
9. Click on a checkbox to select it
10. The value becomes 1
11. Uncheck the box and the value becomes 0
12. Check and uncheck more boxes to see the number increases and decreases

### CODE REVIEW

File: media/system/js/core-uncompressed.js [hunk 1]
```diff
@@ -220,7 +220,7 @@ Joomla.editors.instances = Joomla.editors.instances || {};
 			form = document.getElementById( 'adminForm' );
 		}
 
-		form.boxchecked.value += isitchecked ? 1 : -1;
+		form.boxchecked.value = isitchecked ? parseInt(form.boxchecked.value) + 1 : parseInt(form.boxchecked.value) + -1;
```
Comment (Fedik) [inline media/system/js/core-uncompressed.js]: can be just `parseInt(form.boxchecked.value) - 1` instead of `parseInt(form.boxchecked.value) + -1` :wink:

**Comment by roland-d** [comment]
@Fedik Correct, missed that one. Fixed :)

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 223cbac3c0a987edf573ac973e2ad72733aa82ec<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8590">issues.joomla.org/joomla-cms/8590</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 223cbac3c0a987edf573ac973e2ad72733aa82ec<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8590">issues.joomla.org/joomla-cms/8590</a>.</sub>

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 223cbac3c0a987edf573ac973e2ad72733aa82ec<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8590">issues.joomla.org/joomla-cms/8590</a>.</sub>

**Comment by brianteeman** [comment]
Thanks for testing - setting RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8590">issues.joomla.org/joomla-cms/8590</a>.</sub>

**Pull request merged**